include/picongpu/plugins/radiation/Radiation.kernel

```diff
@@ -56,430 +58,482 @@
+         * @param radiation
+         * @param globalOffset
+         * @param currentStep
+         * @param mapper
+         * @param freqFkt
+         * @param simBoxSize
          */
+        template<
+            typename ParBox,
+            typename DBox,
+            typename Mapping,
+            typename T_Acc
+        >
+        DINLINE
+        /*__launch_bounds__(256, 4)*/
+        void operator()(
+            T_Acc const & acc,
+            ParBox pb,
+            DBox radiation,
+            DataSpace<simDim> globalOffset,
+            uint32_t currentStep,
+            Mapping mapper,
+            radiation_frequencies::FreqFunctor freqFkt,
+            DataSpace<simDim> simBoxSize
+        ) const
+        {
+            using namespace mappings::threads;
+            namespace po = boost::program_options;
 
-        const int blockSize=pmacc::math::CT::volume<Block>::type::value;
-        // vectorial part of the integrand in the Jackson formula
-        PMACC_SMEM( acc, real_amplitude_s, memory::Array< vector_64, blockSize > );
+            constexpr uint32_t frameSize = pmacc::math::CT::volume< SuperCellSize >::type::value;
+            constexpr uint32_t numWorker = T_numWorkers;
 
-        // retarded time
-        PMACC_SMEM( acc, t_ret_s, memory::Array< picongpu::float_64, blockSize > );
+            using FrameType = typename ParBox::FrameType ;
+            using FramePtr = typename ParBox::FramePtr;
 
-        // storage for macro particle weighting needed if
-        // the coherent and incoherent radiation of a single
-        // macro-particle needs to be considered
-        PMACC_SMEM( acc, radWeighting_s, memory::Array< float_X, blockSize > );
+            using namespace parameters; // parameters of radiation
 
-        // particle counter used if not all particles are considered for
-        // radiation calculation
-        PMACC_SMEM( acc, counter_s, int );
+            uint32_t const workerIdx = threadIdx.x;
 
-        PMACC_SMEM( acc, lowpass_s, memory::Array< NyquistLowPass, blockSize > );
+            /// calculate radiated Amplitude
+            /* parallelized in 1 dimensions:
+             * looking direction (theta)
+             * (not anymore data handling)
+             * create shared memory for particle data to reduce global memory calls
+             * every thread in a block loads one particle and every thread runs
+             * through all particles and calculates the radiation for one direction
+             * for all frequencies
+             */
+            constexpr int blockSize = pmacc::math::CT::volume<SuperCellSize>::type::value;
 
+            // vectorial part of the integrand in the Jackson formula
+            PMACC_SMEM( acc, real_amplitude_s, memory::Array< vector_64, blockSize > );
 
-        const int theta_idx = blockIdx.x; //blockIdx.x is used to determine theta
-        const uint32_t linearThreadIdx = threadIdx.x; // used for determine omega and particle id
```

Comment:
Question:
The second palatalization range was removed (`linearThreadIdx`) - how does this work?

Reply:
It now called: `linearIdx` and is defined below.